.github/workflows/dependabot-pr.yml

```diff
@@ -0,0 +1,44 @@
+name: Dependabot PR
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  dependabot:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    if: github.triggering_actor == 'dependabot[bot]'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.PAT }}
+
+      - name: update code
+        run: |
+          input="${{ github.event.pull_request.title }}"
+
+          regex="[Bb]ump (.*) from (.*) to (.*)"
+          if [[ ! "$input" =~ $regex ]]; then
+            echo "unexpected pattern"
+            exit 1
+          fi
+
+          crate_name="${BASH_REMATCH[1]}"
+          old_version="${BASH_REMATCH[2]}"
+          new_version="${BASH_REMATCH[3]}"
+          echo "crate_name: $crate_name, old_version: $old_version, new_version: $new_version"
+
+          ./scripts/cargo-for-all-lock-files.sh -- update -p $crate_name:$old_version --precise $new_version
+          ./scripts/cargo-for-all-lock-files.sh -- tree
+
+      - name: push
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+          git add .
```

Comment:
I'd prefer globing like `:**/Cargo.lock` as before.

anyone can edit `.gitignore`, a bit of extreme thought. but, if we're ever tricked into writing credentials into some random file, it's over.

Reply:
yeah... you're right. let me test it a bit

Reply:
fixed: https://github.com/anza-xyz/agave/pull/1466/commits/0207a164f03255ee445647b4c9bc1af98937279a